src/twisted/internet/interfaces.py

```diff
@@ -1588,9 +1588,6 @@ class IFileDescriptor(ILoggingContext):
 
     def fileno() -> object:
         """
-        @raise: If the descriptor no longer has a valid file descriptor
-            number associated with it.
```

Comment:
It seems like a deficiency in the interface that there is no exception type defined for this. Oh well.

Reply:
None of the implementations within Twisted raise an exception though. Some return -1 if the descriptor is no longer valid.